llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -6574,7 +6611,7 @@ BoUpSLP::TreeEntry::EntryState BoUpSLP::getScalarsVectorizationState(
   case Instruction::ExtractElement: {
     bool Reuse = canReuseExtract(VL, VL0, CurrentOrder);
     // FIXME: Vectorizing is not supported yet for non-power-of-2 ops.
-    if (!has_single_bit(VL.size()))
+    if (!hasFullVectorsOnly(*TTI, VL0->getType(), VL.size()))
```

Comment:
Again, have you audited the exposed code?

Reply:
Reverting it for now